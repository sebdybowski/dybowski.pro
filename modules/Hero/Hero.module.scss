@use 'styles/variables.scss' as *;

.Hero {
    background-image: $brand-bg-inverse;
    position: relative;

    &::before {
        content: '';
        width: 100%;
        height: 100%;
        position: absolute;
        opacity: 0.75;
        background-image: url('/seb.webp');
        background-size: contain;
        background-repeat: no-repeat;
        mix-blend-mode: hard-light;
        background-position: 20vw 10vh;

        @media screen and (max-width: ($breakpoint-md - 1)) {
            background-size: cover;
            background-position: 55% 10vh;
        }

        @media screen and (min-width: $breakpoint-lg) {
            background-position: 35vw 10vh;
        }

    }
}

.Hero__Column {
    color: $text-color-light;

    @media screen and (max-width: ($breakpoint-md - 1)) {
        align-items: center;
        text-align: center;
        justify-content: flex-end;
        margin-bottom: 5vw;
    }
}

.Button__Group {
    display: flex;
    flex-direction: row;
}
