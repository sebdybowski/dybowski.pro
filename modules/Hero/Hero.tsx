import { faLinkedin, faGithub } from '@fortawesome/free-brands-svg-icons';
import React, { FC } from 'react';

import { Button, Column, Container, Heading } from '../../components';

import styles from './Hero.module.scss';

interface HeroProps {}

const Hero: FC<HeroProps> = () => (
  <div className={styles.Hero} data-testid="Hero">
    <Container section>
      <Column vCenter half className={styles.Hero__Column}>
        <Heading level={2}>{'Hi, I\'m Seb'}</Heading>
        <Heading level={1} serif brand>Software&nbsp;Engineer &<br/>UI/UX&nbsp;Designer</Heading>
        <Heading level={2}>Nice to meet you!</Heading>
        <div className={styles.Button__Group}>
          <a href="https://www.linkedin.com/in/sebastian-dybowski/" target="_blank" rel="noreferrer">
            <Button
              icon={faLinkedin}
              outline
              inverted
            >my experience</Button>
          </a>
          <a href="https://github.com/sebdybowski">
            <Button
              icon={faGithub}
              outline
              inverted
            >my code</Button>
          </a>
        </div>
      </Column>
    </Container>
  </div>
);

export default Hero;
