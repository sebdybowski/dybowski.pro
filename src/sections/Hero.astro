---
import SebSm from "../assets/img/seb-sm.webp";
import SebMd from "../assets/img/seb-md.webp";
import SebLg from "../assets/img/seb-lg.webp";
import SebXl from "../assets/img/seb-xl.webp";
import { HeartHandshake, Mouse, Hand, Mail } from "lucide-preact";
---

<header class="container-hybrid" id="hero">
  <div class="grid">
    <div class="info-section">
      <hgroup>
        <h1>Hey there, I'm <strong class="gradient-mark">Seb!</strong></h1>
        <h2>
          With expertise in software development, team leadership, and UX
          design, I have the skills and experience to bring your vision to life!
        </h2>
      </hgroup>
      <a
        href="https://pl.linkedin.com/in/sebastian-dybowski"
        role="button"
        target="_blank">
        <HeartHandshake /> hire me
      </a>
      <a
        href="mailto:dybowski.seb@gmail.com"
        role="button"
        class="secondary"
      >
        <Mail /> email me
      </a>
      <p>
        <i class="bi bi-mouse3 mouse"></i>
        <i class="bi bi-hand-index swipe"></i>
        <strong class="scroll__label">Scroll</strong>
        <strong class="swipe__label">Swipe</strong> for next sections.
      </p>
    </div>
    <div class="image-wrapper">
      <img
        alt="Seb's portrait"
        srcset={`
          ${SebSm.src} 576w,
          ${SebMd.src} 768w,
          ${SebLg.src} 992w,
          ${SebXl.src} 1200w
        `}
      />
    </div>
  </div>
</header>

<style>
  #hero {
    display: flex;
    justify-content: center;
    align-items: center;
    min-height: 85vh;
    min-height: 85svh;
    position: relative;
    overflow: hidden;
  }

  .mouse,
  .swipe {
    margin-top: -0.25rem;
    position: relative;
    animation: MoveUpDown 1s linear infinite;
  }

  .swipe,
  .swipe__label {
    display: none;
  }

  .gradient-mark {
    background-image: var(--primary-gradient);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    background-size: 100% 200%;
    background-position-y: 100%;
    animation: pulse 2s ease-in-out infinite;
  }

  .gradient {
    background-image: var(--primary-gradient);
    background-size: 100% 200%;
    background-position-y: 100%;
    animation: pulse 2s ease-in-out infinite;
    border-radius: 50%;
    position: relative;
  }

  .info-section {
    z-index: 1;
  }

  h1 {
    font-size: 3rem;
  }

  img {
    position: absolute;
    bottom: 0;
    left: 40%;
    height: 100%;
    mix-blend-mode: luminosity;
    opacity: 0.9;
  }

  p {
    margin: 2rem 0;
  }

  @media screen and (max-width: 1200px) {
    .info-section {
      width: 120%;
    }
  }

  @media (max-width: 992px) {
    #hero {
      min-height: auto;
    }

    .grid {
      display: flex;
      flex-direction: column-reverse;
      position: relative;
      justify-items: center;
      align-items: center;
    }

    img {
      position: static;
      opacity: 1;
      margin: 1rem auto;
      width: 70%;
      border-radius: 50%;
      background-image: var(--primary-gradient);
    }

    .image-wrapper {
      width: 100%;
      display: flex;
    }

    .info-section {
      width: 100%;
      text-align: center;
    }

    .mouse,
    .scroll__label {
      display: none;
    }

    .swipe,
    .swipe__label {
      display: inline-block;
    }
  }

  @media screen and (max-width: 576px) {
    .info-section {
      width: 100%;
    }

    h1 {
      font-size: 2.25rem;
    }
  }
</style>

<style is:global>
  mark:not(.gradient-mark) {
    border-radius: 0.25rem;
    margin: 0 0.25rem;
  }
</style>
