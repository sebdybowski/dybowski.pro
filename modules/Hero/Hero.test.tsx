import React from 'react';
import { render } from '@testing-library/react';

import Hero from './Hero';

describe('<Hero />', () => {
  test('it should mount', () => {
    const { asFragment } = render(<Hero />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
