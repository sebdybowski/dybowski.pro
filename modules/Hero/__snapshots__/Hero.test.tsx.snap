// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> it should mount 1`] = `
<DocumentFragment>
  <div
    class="Hero"
    data-testid="Hero"
  >
    Hero Component
  </div>
</DocumentFragment>
`;
